torch/_inductor/codegen/multi_kernel.py

```diff
@@ -478,36 +553,64 @@ def _get_cached_shape_choice(self, cache_key):
 
     def _cache_shape_choice(self, cache_key, kernel_idx):
         """
-        Cache kernel choice for a specific shape
+        Cache kernel choice for a specific shape.
         """
         self._shape_cache[cache_key] = kernel_idx
 
+    def _l1_dist(self, k1, k2):
```

Comment:
this name seems misleading? this is more like a custom log-domain heuristic?

in that vein, maybe you can also introduce some other heuristics like true euclidean distance l1 and allow users to override the heuristic strategy?